File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        my_dict = {0 : 1}
        res = 0
        cursum = 0
        for i in range(len(nums)):
            cursum += nums[i]
            if cursum - k in my_dict:
                res += my_dict[cursum - k]
            if cursum in my_dict:
                my_dict[cursum] += 1
            else:
                my_dict[cursum] = 1
        return res

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int len = t.length();
        int z = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s[i] == t[z]) {
                z++;
            }
        }
        return len - z;
    }
};